src/Cache.ts

```diff
@@ -14,13 +14,21 @@ const defaultCacheRoot = envPaths('electron', {
 export class Cache {
   constructor(private cacheRoot = defaultCacheRoot) {}
 
-  public getCachePath(downloadUrl: string, fileName: string): string {
+  public static getCacheDirectory(downloadUrl: string): string {
+    const parsedDownloadUrl = url.parse(downloadUrl);
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    const { search, hash, ...rest } = url.parse(downloadUrl);
-    const strippedUrl = url.format(rest);
+    const { search, hash, pathname, ...rest } = parsedDownloadUrl;
+    const strippedUrl = url.format({ ...rest, pathname: path.dirname(pathname || 'electron') });
+
+    return crypto
+      .createHash('sha256')
+      .update(strippedUrl)
+      .digest('hex')
+      .substr(0, 10);
```

Comment:
I hope we don't hit any collisions with 10 characters. The main file I'm worried about is the checksum file.

Reply:
We still get `16^10`, or `1099511627780` different checksums. For two different URLs to match they have to match the same checksum *and* the same file name. I think we'll be fine. 😉

Reply:
16^10 is the problem space, sure, but the hash algorithm could somehow favor one prefix over another (because overall, the hash is "unique", but prefixes aren't as guaranteed).

The checksum file does have the same name for every version, that's why I called it out.

Reply:
I meant `electron-v9.4.2-win32-arm64.zip` as a file name, not the checksum. I'd love to get this in asap to address issues downstream. So if this stripping isn't good enough let me know and I can revert it.